pkgs/games/devilutionx/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchgit, cmake, gcc, SDL2, SDL2_mixer, SDL2_ttf, libsodium, pkg-config }:
+stdenv.mkDerivation rec {
+  version = "0.5.0-SNAPSHOT";
+  name = "devilutionx-${version}";
+  rev = "b2f358874705598ec139f290b21e340c73d250f6"; # current master
+
+  src = fetchgit {
+    url = "https://github.com/diasurgical/devilutionX.git";
+    inherit rev;
+    sha256 = "0s812km118qq5pzlzvzfndvag0mp6yzvm69ykc97frdiq608zw4f";
+  };
+
+  NIX_CFLAGS_COMPILE = "-I${SDL2_ttf}/include/SDL2";
```

Comment:
Nether `cmake` nor `pkgconfig` know what to do here? Is that an upstream issue?

Reply:
I was getting this error:
```
[ 74%] Building CXX object CMakeFiles/devilution.dir/Source/wave.cpp.o
[ 75%] Linking CXX static library libdevilution.a
[ 75%] Built target devilution
Scanning dependencies of target devilutionx
[ 76%] Building CXX object CMakeFiles/devilutionx.dir/SourceX/miniwin/misc.cpp.o
[ 76%] Building CXX object CMakeFiles/devilutionx.dir/SourceX/dx.cpp.o
[ 77%] Building CXX object CMakeFiles/devilutionx.dir/SourceX/miniwin/misc_io.cpp.o
[ 78%] Building CXX object CMakeFiles/devilutionx.dir/SourceX/miniwin/misc_msg.cpp.o
In file included from /build/source/SourceX/miniwin/misc.cpp:7:0:
/build/source/SourceX/DiabloUI/diabloui.h:4:10: fatal error: SDL_ttf.h: No such file or directory
 #include <SDL_ttf.h>
          ^~~~~~~~~~~
compilation terminated.
```

I've searched a bit in other packages and I've come up with this solution... I don't use C++, but it seems like the header of SDL2_ttf was not seen by the compiler, and this flag solves it. 

I'd be happy to use a different solution.

Reply:
```
~/nixpkgs> nix-build -A SDL2_ttf
/nix/store/s3x7vqpw2nbxgnqyq31q6562bsygv6wi-SDL2_ttf-2.0.15

~/nixpkgs> ls /nix/store/s3x7vqpw2nbxgnqyq31q6562bsygv6wi-SDL2_ttf-2.0.15/lib/pkgconfig/
SDL2_ttf.pc

~/nixpkgs> cat /nix/store/s3x7vqpw2nbxgnqyq31q6562bsygv6wi-SDL2_ttf-2.0.15/lib/pkgconfig/SDL2_ttf.pc
prefix=/nix/store/s3x7vqpw2nbxgnqyq31q6562bsygv6wi-SDL2_ttf-2.0.15
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: SDL2_ttf
Description: ttf library for Simple DirectMedia Layer with FreeType 2 support
Version: 2.0.15
Requires: sdl2 >= 2.0.8
Libs: -L${libdir} -lSDL2_ttf
Cflags: -I${includedir}/SDL2
```

Definitely something funky going on here... digging a bit deeper, stand by.

Reply:
Confirmed `SDL2_ttf` is not being included at all using `cmakeFlags = [ "-DCMAKE_VERBOSE_MAKEFILE=ON" ];`:

`/nix/store/hpzj855nkgjvg58nrhq4910sb9q3kss1-gcc-wrapper-7.4.0/bin/g++  -DASIO_STANDALONE -DDEVILUTION_STUB -DFASTER -D_DEBUG -I/build/source/SourceX -I/build/source/3rdParty/asio/include -I/build/source/3rdParty/Radon/Radon/include -I/build/source/3rdParty/libsmacker -I/build/source/Source -I/build/source/SourceS -isystem /nix/store/nrhxlhmrnrvzsj43jia7b75gh5yc6gpg-SDL2-2.0.9-dev/include/SDL2 -isystem /nix/store/gyaz86rz0rxznvphs36cjl86ms4bzqw5-SDL2_mixer-2.0.4/include/SDL2  -O3 -DNDEBUG   -Wall -Wextra -Wno-write-strings -Wno-multichar -Wno-unused-parameter -fsigned-char -fsanitize=address -fsanitize-recover=address -Wno-unknown-pragmas -pthread -std=c++14 -o CMakeFiles/devilutionx.dir/SourceX/miniwin/dsound.cpp.o -c /build/source/SourceX/miniwin/dsound.cpp`

Reply:
Maybe we should create a new issue regarding `SDL2_ttf` ?

Reply:
As you may have noticed I posted the question on discourse but currently have no reply. An issue sounds like a good idea.

I'm inclined to merge this without a solution art this time with the hope that a solution will be found and then the package updated.